### PR TITLE
added method extractAllToAsync

### PR DESCRIPTION
**Pull request by abner**
Added method extractAllToAsync

### CODE REVIEW

**Comment by jvilk** [comment]
@abner did you change the line endings in `.gitignore` or something? you should omit that from your commit.

The function you added [isn't documented in `adm-zip`'s documentation](https://github.com/cthackers/adm-zip/wiki/ADM-ZIP), but I hand-verified that it exists. Can you remove the newlines around the function definition so it stands correctly with the others?

Once those two things are done, let me know and I'll sign off on it.

**Comment by abner** [comment]
ok, thanks @jvilk. i just reverted the .gitignore changes and removed the new lines.

File: adm-zip/adm-zip.d.ts [hunk 1]
```diff
@@ -212,6 +212,15 @@ declare module "adm-zip" {
...
         extractAllTo(targetPath: string, overwrite?: boolean): void;
         /**
+         * Extracts the entire archive to the given location
+         * @param targetPath Target location
+         * @param overwrite If the file already exists at the target path, the file
+         *   will be overwriten if this is true. Default is FALSE
+         * @param callback The callback function will be called afeter extraction 
+         */
+        extractAllToAsync(targetPath: string, overwrite: boolean, callback: (error: Error) => void): void;
+
```
Comment (jvilk) [inline adm-zip/adm-zip.d.ts]: I actually meant this line... could you remove it?

Sorry for being unclear!

Comment (abner) [inline adm-zip/adm-zip.d.ts]: Ok. Just did it.

**Comment by jvilk** [comment]
:+1:

**Pull request merged**